src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -594,7 +594,7 @@ public static String generateSASURL(String storageAccountName, String storageAcc
 				  new URI(getCustomURI(storageAccountName, TABLE, saBlobURI)));
 		// Create the blob client.
 		CloudBlobClient blobClient = cloudStorageAccount.createCloudBlobClient();
-		CloudBlobContainer container = blobClient.getContainerReference(containerName);
+		CloudBlobContainer container = blobClient.getContainerReference(containerName.substring(1));
```

Comment:
We remove the first character of the container name, because the leading '/' makes the container object throw a storageException.